src/app/work-item/work-item-board/work-item-board.component.ts

```diff
@@ -90,6 +90,11 @@ export class WorkItemBoardComponent implements OnInit, OnDestroy {
     private areaService: AreaService,
     private filterService: FilterService,
     private route: ActivatedRoute) {
+      router.events.subscribe(event => {
```

Comment:
* I believe, every time you open the details page this will trigger even if you are not adding a new item. In fact on every navigation the list will re load with this chunk of code.  
* Also it's not being unsubscribed anywhere, so that could cause a event leakage.